src/Credentials.js

```diff
@@ -141,98 +309,33 @@ class Credentials {
+  *
+  *  The main difference between this and `authenticate()` is that it does not verify the challenge. This can be used to verify user profiles that have been shared
+  *  through other methods such as QR codes and messages.
   *
-  *  @param    {String}                  token              a push notification token (get a pn token by requesting push permissions in a request)
-  *  @param    {Object}                  payload            push notification payload
-  *  @param    {String}                  payload.url        a uport request url
-  *  @param    {String}                  payload.message    a message to display to the user
-  *  @param    {String}                  pubEncKey          the public encryption key of the receiver, encoded as a base64 string
-  *  @return   {Promise<Object, Error>}              a promise which resolves with successful status or rejects with an error
+  *  @example
+  *  const resToken = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJyZXF1Z....'
+  *  credentials.verifyProfile(resToken).then(profile => {
+  *      const credentials = profile.verified
+         const name =  profile.name
+  *      ...
+  *  })
+  *
+  *  @param    {String}                  token                 a response token
+  *  @return   {Promise<Object, Error>}                        a promise which resolves with a parsed response or rejects with an error.
   */
-  push (token, pubEncKey, payload) {
-    const PUTUTU_URL = 'https://pututu.uport.space' // TODO - change to .me
-    return new Promise((resolve, reject) => {
-      let endpoint = '/api/v2/sns'
-      if (!token) {
-        return reject(new Error('Missing push notification token'))
-      }
-      //if (!pubEncKey) {
-        //return reject(new Error('Missing public encryption key of the receiver'))
-      //}
-      if (pubEncKey.url) {
-        console.error('WARNING: Calling push without a public encryption key is deprecated')
-        endpoint = '/api/v1/sns'
-        payload = pubEncKey
-      } else {
-        if (!payload.url) {
-          return reject(new Error('Missing payload url for sending to users device'))
-        }
-        const plaintext = padMessage(JSON.stringify(payload))
-        const enc = encryptMessage(plaintext, pubEncKey)
-        payload = { message: JSON.stringify(enc) }
-      }
-
-      nets({
-        uri: PUTUTU_URL + endpoint,
-        json: payload,
-        method: 'POST',
-        withCredentials: false,
-        headers: {
-          Authorization: `Bearer ${token}`
-        }
-      },
-      (error, res, body) => {
-        if (error) return reject(error)
-        if (res.statusCode === 200) {
-          resolve(body)
-        }
-        if (res.statusCode === 403) {
-          return reject(new Error('Error sending push notification to user: Invalid Token'))
-        }
-        reject(new Error(`Error sending push notification to user: ${res.statusCode} ${body.toString()}`))
-      })
-    })
-  }
+ async verifyProfile (token) {
```

Comment:
👍  yeah this function is needed, but the name if very confusing to me, while its main purpose is to still verify parse and return parsed response, verifyProfile seems to mean something different, like it would consume a profile or something, not a response

Reply:
This is the profile received from a contact, as opposed to logging in a user